lib/connectors/sentinel_connector.js

```diff
@@ -36,7 +36,10 @@ SentinelConnector.prototype.connect = function (callback) {
     this.currentPoint = -1;
   }
   if (!Array.isArray(this.sentinels)) {
-    this.sentinels = this.options.sentinels;
+    this.sentinels = this.options.sentinels.map(function (endpoint) {
+      endpoint.port = endpoint.port || 26379;
+      return endpoint;
+    });
```

Comment:
Updating properties of the option introducing side effects. For example:

```javascript
const sentinels: [{host: '192.168.1.3'}]
const redis = new Redis({sentinels})

console.log(sentinels) // values changed unexpectedly
```

What about give the default port when creating a new instance by `port: endpoint.port || 26379`: https://github.com/luin/ioredis/blob/master/lib/connectors/sentinel_connector.js#L213.

There's an `isSentinelEql` function should be updated accordingly.